provider/client.go

```diff
@@ -75,6 +75,11 @@ func NewClient(host, env, tenant, clientId, clientSecret, region *string) (*Clie
 	}
 
 	c.Token = ar.Token
+	scheduledTime := time.Duration(ar.ExpiresIn)*time.Second - 2*time.Minute
```

Comment:
Here the `scheduledTime` could be < 0 if we ever change the validity of a token to < 2 minutes but I suppose this is very unlikely